lib/pbench/server/filetree.py

```diff
@@ -340,9 +354,27 @@ def unpack(self, incoming: Path, results: Path):
                     f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
                 )
 
-        self.unpacked = unpacked
+        command = "find unpacked/ \( -type d -exec chmod ugo+rx '{}' \; \) -o \( -type f -exec chmod ugo+r '{}' \; \)"
+        try:
+            process = subprocess.run(
+                command,
+                shell=True,
+                stdin=subprocess.DEVNULL,
+                capture_output=True,
+                text=True,
+            )
+        except Exception as exc:
+            raise TarballModeChangeError(self.tarball_path.name, str(exc)) from exc
+        else:
+            if process.returncode != 0:
+                raise TarballModeChangeError(
+                    self.tarball_path.name,
+                    f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
+                )
+
+        self.unpacked = Path(str(unpacked)[: -len(".unpack")])
         results_link = results / self.name
-        results_link.symlink_to(unpacked)
+        results_link.symlink_to(self.unpacked)
```

Comment:
You want to `mv` (rename) the temporary directory, not just symlink it, once you're done. (That is, we don't want to end up with *both* names in the INCOMING directory.)

Reply:
I pushed the changes a little too early 😄

Reply:
Shall we rename tarball.unpack directory to tarball? I think that would be also a good idea. What do you think?

Reply:
GitHub is being weird this morning. I can see email showing that you made a comment here,

> Shall we rename tarball.unpack directory to tarball? I think that would be also a good idea. What do you think?

But GitHub won't show it to me either in the files view or on the conversation tab, no matter how many times I reload. (I even restarted Chrome.) So maybe the whole universe this morning is a figment of my imagination.

Yes; we have created an `{INCOMING}/{controller}/{tarball}.unpack` directory, and we want to rename ("move") to `{INCOMING}/{controller}/{tarball}` once we know we've prepared it properly.

Reply:
> I can see email showing that you made a comment here, [...] But GitHub won't show it to me

I'm guessing that that means that she subsequently deleted it.  ✂️